kong/plugin_service_test.go

```diff
@@ -107,6 +107,36 @@ func TestPluginWithTags(T *testing.T) {
 	assert.NoError(err)
 }
 
+func TestPluginWithOrdering(T *testing.T) {
+	RunWhenEnterprise(T, ">=3.0.0", RequiredFeatures{})
+	assert := assert.New(T)
+
+	client, err := NewTestClient(nil, nil)
+	assert.NoError(err)
+	assert.NotNil(client)
+
+	plugin := &Plugin{
+		Name: String("request-termination"),
+		Ordering: &PluginOrdering{
+			Before: PluginOrderingPhase{
+				"access": []string{"key-auth", "basic-auth"},
+			},
+		},
+	}
+
+	createdPlugin, err := client.Plugins.Create(defaultCtx, plugin)
+	assert.NoError(err)
+	assert.NotNil(createdPlugin)
+	assert.Equal(PluginOrdering{
+		Before: PluginOrderingPhase{
+			"access": []string{"key-auth", "basic-auth"},
+		},
+	}, *createdPlugin.Ordering)
+
+	err = client.Plugins.Delete(defaultCtx, createdPlugin.ID)
+	assert.NoError(err)
+}
+
```

Comment:
Trying to understand what this is testing. There is no new functionality added in this PR. Are testing marshalling?

Reply:
Essentially. Pretty much everything in go-kong is marshalling and unmarshalling 🙂

Our utilities for version-dependent tests affect the whole test, so this is separate to let it run on 3.0+ only, similar to the separate tag test.

Reply:
Thanks